src/main/java/io/strimzi/kafka/metrics/HttpServers.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics;
+
+import io.prometheus.metrics.exporter.httpserver.HTTPServer;
+import io.prometheus.metrics.model.registry.PrometheusRegistry;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Class to keep track of all the HTTP servers started by all the Kafka components in a JVM.
+ */
+public class HttpServers {
+
+    private static final Map<Listener, ServerCounter> SERVERS = new HashMap<>();
+
+    /**
+     * Get or create a new HTTP server if there isn't an existing instance for the specified listener.
+     * @param listener The host and port
+     * @param registry The Prometheus registry to expose
+     * @return A ServerCounter instance
+     * @throws IOException if the HTTP server does not exist and cannot be started
+     */
+    public synchronized static ServerCounter getOrCreate(Listener listener, PrometheusRegistry registry) throws IOException {
```

Comment:
Yeah the registry used to be hard coded to the default registry as it's what all components have to use. But because there isn't a mechanism to clear a registry, we allowed passing it in for tests.